crates/jstzd/src/task/octez_client.rs

```diff
@@ -0,0 +1,156 @@
+        self.endpoint = Some(endpoint);
+        self
+    }
+    pub fn set_disable_unsafe_disclaimer(
+        mut self,
+        disable_unsafe_disclaimer: bool,
+    ) -> Self {
+        self.disable_unsafe_disclaimer = disable_unsafe_disclaimer;
+        self
+    }
+    pub fn build(mut self) -> OctezClient {
+        let base_dir = self
+            .base_dir
+            .map_or(OctezClientDirectory::default(), OctezClientDirectory::Path);
+        OctezClient {
+            binary_path: self.binary_path.take(),
+            base_dir,
+            endpoint: self.endpoint.unwrap_or_default(),
+            disable_unsafe_disclaimer: self.disable_unsafe_disclaimer,
+        }
+    }
+}
+
+pub enum OctezNodeEndpoint {
+    RpcEndpoint(Endpoint),
+    Node(OctezNode),
+}
+
+impl Default for OctezNodeEndpoint {
+    fn default() -> Self {
+        OctezNodeEndpoint::RpcEndpoint(Endpoint::localhost(8732))
+    }
+}
+
+impl From<Endpoint> for OctezNodeEndpoint {
+    fn from(endpoint: Endpoint) -> Self {
+        OctezNodeEndpoint::RpcEndpoint(endpoint)
+    }
+}
+impl From<OctezNode> for OctezNodeEndpoint {
+    fn from(node: OctezNode) -> Self {
+        OctezNodeEndpoint::Node(node)
+    }
+}
+
+pub enum OctezClientDirectory {
+    TempDir(TempDir),
+    Path(PathBuf),
+}
+
+impl Default for OctezClientDirectory {
+    fn default() -> Self {
+        OctezClientDirectory::TempDir(tempdir().expect("Failed to create tempdir"))
+    }
+}
+
+// TODO: remove later
+#[allow(dead_code)]
+pub struct OctezClient {
+    // if None, use the binary in $PATH
+    binary_path: Option<PathBuf>,
+    base_dir: OctezClientDirectory,
+    endpoint: OctezNodeEndpoint,
+    disable_unsafe_disclaimer: bool,
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    #[test]
+    fn builds_default_octez_client() {
+        let octez_client = OctezClientBuilder::new().build();
+        assert!(octez_client.binary_path.is_none());
+        assert!(matches!(
+            octez_client.base_dir,
+            OctezClientDirectory::TempDir(_)
+        ));
+        assert!(!octez_client.disable_unsafe_disclaimer);
+        match octez_client.endpoint {
+            OctezNodeEndpoint::RpcEndpoint(endpoint) => {
+                assert_eq!(endpoint, Endpoint::localhost(8732))
+            }
+            _ => panic!("Expected RpcEndpoint"),
+        }
+    }
+
+    #[test]
+    fn temp_dir_is_created_by_default() {
+        let octez_client = OctezClientBuilder::new().build();
+
+        let temp_dir = match octez_client.base_dir {
+            OctezClientDirectory::TempDir(temp_dir) => temp_dir,
+            _ => panic!("Expected TempDir"),
+        };
+        assert!(temp_dir.path().exists());
+    }
+
+    #[test]
+    fn temp_dir_is_removed_on_drop() {
+        let octez_client = OctezClientBuilder::new().build();
+
+        let temp_dir_path = match &octez_client.base_dir {
+            OctezClientDirectory::TempDir(temp_dir) => temp_dir.path().to_path_buf(),
```

Comment:
Inline the drop and assert (like above) for consistency

Reply:
done